docs/cloud-tutorials/best-practices.md

```diff
@@ -0,0 +1,87 @@
+
+## Managing applications
+
+### Registration
+* An application must be registered before being deployed. DBOS Cloud CLI will handle registration by default when you deploy an application for the first time.
+* Only one request at a time can manipulate an application and you will get an `application is busy` error if you attempt multiple concurrent modifications.
+* Application names are unique within an [organization](#organizations-and-users-management)
+
+### Deployment
+* Interrupting a deployment request - for example, pressing control + C - will *not* cancel the deploy action on DBOS Cloud
+* DBOS Cloud will prune all development dependencies. Make sure to explicitly install any runtime dependency you need with `npm install`
+* Your application must not use more than 1GB of disk space
+* DBOS Cloud stores each version of your application. You can download a specific version using the DBOS Cloud CLI [TOLINK WHEN MERGED]
+
+### During runtime
+* Nothing on the filesystem is persisted between deployments
+* DBOS Cloud will keep around VMs running old versions of a workflow. This means, you need to be mindful of non-backward compatible database schema changes.
+
+### Deletion
+* Deleting an app is an asynchronous process and there is a small delay between a deletion request completing and the application being fully deleted. This means you could hit an "application is busy" error when re-deploying an application you just deleted
+* An application will be deleted once all PENDING workflows -- including these previous versions -- are completed
+* All application archives -- including previous versions -- will be deleted
+* You can elect to drop the application database alongside deleting an application. This will also delete all the database roles associated with the application in the database instance.
+
+### Time travel
+You cannot enable time travel on an existing application. You will need to register a new application.
+See more information in the [time travel](#time-travel) section.
+
+### Swapping database instance
+While an application cannot change database once registered, you can change the database instance (server) hosting the application database. When doing so, you can also decide to redeploy a previous version of the application. This is how DBOS Cloud support point in time recovery today.
+
+## Application databases
+
+:::tip
+Application databases are "logical" postgres databases. They are not to be confused with database *instances*, the server hosting application databases.
+:::
+
+* Application databases cannot be shared across applications
+* An application cannot change its database once deployed
+
+### Database roles
+
+* DBOS Cloud will create the following roles for each application:
+    * `dbosadmin` - a privileged role used to execute your schema migration steps during application deployment
+    * `approle` - a user-provided role you provide when provisioning the database instance. This role has
```

Comment:
The name of this role can either be user provided, or by default `dbos_user` (https://github.com/dbos-inc/dbos-transact/pull/598)